test/e2e/e2e.go

```diff
@@ -629,25 +629,25 @@ spec:
 
 		framework.Logf("Applying the EgressIP configuration")
 		framework.RunKubectlOrDie("apply", "-f", egressIPYaml)
-		time.Sleep(waitInterval)
 
-		targetExternalContainerAndTest := func(verifyIPType, podName string, verifyIPs []string) {
-			framework.RunKubectlOrDie("exec", podName, frameworkNsFlag, "--", "curl", net.JoinHostPort(targetNode.nodeIP, "80"))
-			targetNodeLogs, err := runCommand("docker", "logs", targetNode.name)
-			if err != nil {
-				framework.Failf("failed to inspect logs in test container: %v", err)
-			}
-			targetNodeLogs = strings.TrimSuffix(targetNodeLogs, "\n")
-			logLines := strings.Split(targetNodeLogs, "\n")
-			lastLine := logLines[len(logLines)-1]
-			var found bool
-			for _, verifyIP := range verifyIPs {
-				if strings.Contains(lastLine, verifyIP) {
-					found = true
+		targetExternalContainerAndTest := func(verifyIPType, podName string, verifyIPs []string) wait.ConditionFunc {
+			return func() (bool, error) {
+				framework.RunKubectlOrDie("exec", podName, frameworkNsFlag, "--", "curl", net.JoinHostPort(targetNode.nodeIP, "80"))
+				targetNodeLogs, err := runCommand("docker", "logs", targetNode.name)
+				if err != nil {
+					framework.Logf("failed to inspect logs in test container: %v", err)
+					return false, nil
 				}
-			}
-			if !found {
-				framework.Failf("the test external container did not have any trace of the %s IPs: %s being logged, last logs: %s", verifyIPType, verifyIPs, logLines[len(logLines)-1])
+				targetNodeLogs = strings.TrimSuffix(targetNodeLogs, "\n")
+				logLines := strings.Split(targetNodeLogs, "\n")
+				lastLine := logLines[len(logLines)-1]
+				for _, verifyIP := range verifyIPs {
+					if !strings.Contains(lastLine, verifyIP) {
+						framework.Logf("the test external container did not have any trace of the %s IPs: %s being logged, last logs: %s", verifyIPType, verifyIPs, logLines[len(logLines)-1])
+						return false, nil
```

Comment:
I think you only want to return false if the log doesnt have *any* of the verifyIPs:

2021-02-09T23:54:18.9799592Z Feb  9 23:54:18.979: INFO: the test external container did not have any trace of the egress IPs: [172.18.1.4 172.18.1.5] being logged, last logs: 172.18.1.4 - - [09/Feb/2021:23:54:18 +0000] "GET / HTTP/1.1" 200 45

^ you see there it contained 1 of them, but this for loop will return false if any of the verifyIPs are not found

Reply:
@nerdalert or @alexanderConstantinescu can correct me if I'm wrong here

Reply:
Yes, that is correct

Reply:
cool, I wasn´t sure , so it is ANY